src/coord/dist_aggregate.c

```diff
@@ -123,7 +123,8 @@ static int netCursorCallback(MRIteratorCallbackCtx *ctx, MRReply *rep) {
 
   // rewrite and resend the cursor command if needed
   int rc = REDIS_OK;
-  bool done = !getCursorCommand(rep, cmd, MRIteratorCallback_GetCtx(ctx));
+  // done should only be determined based on the cursor and not on the set of results we get
+  const bool done = !getCursorCommand(rep, cmd, MRIteratorCallback_GetCtx(ctx));
```

Comment:
consider moving the call to this function to where we check `done`

Reply:
getCursorCommand modifies cmd contents internally
So currently it has to happen before the if block.(since the if block accesses cmd)

Reply:
But notice that none of the changes are relevant to what we checked there, And also if it returns false (`done` is true) we don't change it